src/python/pants/source/filespec.py

```diff
@@ -4,69 +4,20 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import re
-
-
-def glob_to_regex(pattern):
-  """Given a glob pattern, return an equivalent regex expression.
-
-  TODO: Replace with implementation in `fs.rs`. See https://github.com/pantsbuild/pants/issues/6795.
-
-  :param string glob: The glob pattern. "**" matches 0 or more dirs recursively.
-                      "*" only matches patterns in a single dir.
-  :returns: A regex string that matches same paths as the input glob does.
-  """
-  out = ['^']
-  components = pattern.strip('/').replace('.', '[.]').replace('$','[$]').split('/')
-  doublestar = False
-  for component in components:
-    if len(out) == 1:
-      if pattern.startswith('/'):
-        out.append('/')
-    else:
-      if not doublestar:
-        out.append('/')
-
-    if '**' in component:
-      if component != '**':
-        raise ValueError('Invalid usage of "**", use "*" instead.')
-
-      if not doublestar:
-        out.append('(([^/]+/)*)')
-        doublestar = True
-    else:
-      out.append(component.replace('*', '[^/]*'))
-      doublestar = False
-
-  if doublestar:
-    out.append('[^/]*')
-
-  out.append('$')
-
-  return ''.join(out)
+from pants.engine.fs import PathGlobs
+from pants.engine.native import Native
 
 
 def globs_matches(paths, patterns, exclude_patterns):
-  def excluded(path):
-    if excluded.regexes is None:
-      excluded.regexes = [re.compile(glob_to_regex(ex)) for ex in exclude_patterns]
-    return any(ex.match(path) for ex in excluded.regexes)
-  excluded.regexes = None
-  for pattern in patterns:
-    regex = re.compile(glob_to_regex(pattern))
-    for path in paths:
-      if regex.match(path) and not excluded(path):
-        return True
-  return False
+  path_globs = PathGlobs(include=patterns, exclude=exclude_patterns)
+  return Native().match_path_globs(path_globs, paths)
```

Comment:
I didn't realize you could just construct a `Native()` in any random python code and it's giving me ideas!

Reply:
It got much easier recentlyish: #6979